pkg/api/addTask.go

```diff
@@ -0,0 +1,70 @@
+package server
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/muradrmagomedov/final-project/pkg/services"
+	"github.com/muradrmagomedov/final-project/pkg/todo"
+)
+
+func addTask(w http.ResponseWriter, r *http.Request) {
+
+	const op = "server.addTask"
+	var newTask todo.Task
+
+	body, err := io.ReadAll(r.Body)
+	if err != nil {
+		sendError(w, op, errorReadData, http.StatusInternalServerError)
+		return
+	}
+	err = json.Unmarshal(body, &newTask)
+	if err != nil {
+		sendError(w, op, errorReadData, http.StatusInternalServerError)
+		return
+	}
+	if newTask.Title == "" {
+		msg := errorEmpty + "title"
+		sendError(w, op, msg, http.StatusBadRequest)
+		return
+
+	}
+
+	if newTask.Date == "" {
+		newTask.Date = time.Now().Format("20060102")
+	}
+
+	date, err := time.Parse("20060102", newTask.Date)
+	if err != nil {
+		msg := errorWrongFormat + "date"
+		sendError(w, op, msg, http.StatusBadRequest)
+		return
+	}
+	if date.Format("20060102") < time.Now().Format("20060102") {
+		if newTask.Repeat == "" {
+			date = time.Now()
+			newTask.Date = date.Format("20060102")
+		} else {
+			nextDate, err := services.NextDate(time.Now(), newTask.Date, newTask.Repeat)
+			if err != nil {
+				sendError(w, op, err.Error(), http.StatusBadRequest)
+				return
+			}
+			newTask.Date = nextDate
+		}
+	}
+	id, err := Repo.AddTask(newTask)
+	if err != nil {
+		sendError(w, op, err.Error(), http.StatusBadRequest)
+		return
+	}
+	w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+	answerId, err := json.Marshal(todo.Task{Id: id})
+	if err != nil {
+		sendError(w, op, errorMarshal, http.StatusInternalServerError)
+		return
+	}
+	w.Write(answerId)
```

Comment:
В идеале ошибку записи ответа нужно просто логировать
```suggestion
	_, err := w.Write(answerId)
```